launcher/ui/widgets/ModFilterWidget.cpp

```diff
@@ -249,7 +293,9 @@ void ModFilterWidget::onHideInstalledFilterChanged()
 void ModFilterWidget::onVersionFilterTextChanged(const QString& version)
 {
     m_filter->versions.clear();
-    m_filter->versions.emplace_back(version);
+    if (version != tr("All Versions")) {
```

Comment:
If a translator makes a mistake and translates this even slightly differently then it will break.

I would suggest returning the version ID with Qt::UserRole as well, and returning "all" for the "All Versions" item.